Add details URL for each market

Roshanjossey opened the pull request:
Address #318

I wasn't able to get jasmine running. 

I've only added links for two market entries

![image](https://user-images.githubusercontent.com/8488446/95119298-094ecc00-074c-11eb-8bdb-d07fa43a3984.png)


You can find it in https://deploy-preview-337--wo-ist-markt.netlify.app/#hamburg

mojoaxel reviewed (approved):
LGTM

johnjohndoe commented on cities/hamburg.json:
```diff
@@ -1160,7 +1160,8 @@
                 "title" : "Flohschanze",
-                "location" : "Neuer Kamp 30"
+                "location" : "Neuer Kamp 30",
+                "details_URL" : "https://www.hamburg.de/flohmarkt/4245446/flohschanze/"
```
Please change `details_URL` to `details_url` to have all attributes consistent in lowercase.

Roshanjossey replied:
I'll make this change

Roshanjossey replied:
Done

johnjohndoe commented on css/main.css:
```diff
@@ -146,6 +146,11 @@ html, body {
 }
 
+.leaflet-popup-content a {
+    display: block;
```
Why is `display: block;` needed?

Roshanjossey replied:
I'm using `display: block;` along with `line-height` to increase the touch area for the link

Roshanjossey replied:
anchor tag has `display: inline` by default

johnjohndoe replied:
Ah nice. Thank you for the explanation. Do you think adding the reason as a comment on the attribute is useful for others?

Roshanjossey replied:
Sure. I'll add a comment

johnjohndoe commented on js/main.js:
```diff
@@ -264,14 +264,18 @@ function initMarker(feature) {
         where = '';
     }
+
+    var additionalInformationLink = properties.details_URL ?
+        '<span><a href='+properties.details_URL+'> weitere Informationen </a></span>' : '';
+
```
:+1: I like that no HTML element is added if the attribute is missing. Does it also account for an empty string?

Roshanjossey replied:
Yes

```Boolean('')```
> false

johnjohndoe reviewed (changes requested):
Thank you for working on this new feature.

Besides the comments, I am missing an update on the [contribution guide](https://github.com/wo-ist-markt/wo-ist-markt.github.io/blob/master/CONTRIBUTING.md). The `details_url` attribute should be mentioned there as an optional field.

Please also document in your pull request on which devices and operating systems/versions you tested the feature. Feel free to request someone to complete the tests, e.g. on Android, iPhone, ...

---

Please **rebase** your `feat-318-add-details-url-for-each-market` branch on top of the latest commit on `master`. Let me know if you need help with that.

Roshanjossey commented:
I've added information about `details_url` in `CONTRIBUTING.md`

Also rebased my branch on top of master.

#### Testing

I've tested these changes in 
Firefox(version 79.0 (64-bit)) and Chrome (Version 84.0.4147.125) in Ubuntu 16.04 LTS,
Firefox(version 81.1.2) on Android
Firefox (version 81.0.1) and Chrome(Version 85.0.4183.121) on Windows 10

johnjohndoe reviewed (approved):
@Roshanjossey Thank you for the latest changes. I review and tested them myself and I am happy to approve them now :)

Please note that I squashed your commits into a single one because the later ones are fixups on the initial one. I also rebased your branch onto `master` - your `master` branch was not up-to-date. Please pull from time to time.

# Successfully tested on
- Pixel 2, Android 10, Chrome 85.0.4183.127
- Pixel 2, Android 10, DuckDuckGo 5.66.1 (566601)
- Ubuntu 18.04.4, Brave Version 1.11.104 Chromium: 84.0.4147.105 (Official Build) (64-bit), mobile devices mode

Roshanjossey commented:
Thank you @johnjohndoe.

> Please note that I squashed your commits into a single one because the later ones are fixups on the initial one. I also rebased your branch onto `master` - your `master` branch was not up-to-date. Please pull from time to time.

Sure. I'll keep this in mind.

johnjohndoe commented:
@Roshanjossey It would be great if you could advertise this new feature to all the cities (people maintaining the market data). A good place for this might be our [Slack channel](https://openknowledgegermany.slack.com) or just Twitter addressing `@CodeFor..`.

Roshanjossey commented:
@johnjohndoe, I can't sign up on open knowledge slack team with that link. Could you please send an invite to roshanjossey[at]gmail.com ?

johnjohndoe commented:
@Roshanjossey Invite sent. The project here originated from [Code for Germany](https://codefor.de). It was started in Karlsruhe. Almost every city has a Twitter handle, such as [`@codeforbe`](http://twitter.com/codeforbe).
